disco/bot/bot.py

```diff
@@ -327,10 +346,17 @@ def get_commands_for_message(self, require_mention, mention_rules, prefix, msg):
 
             content = content.lstrip()
 
-        if prefix and not content.startswith(prefix):
-            return []
+        # Scan through the prefixes to find the first one that matches.
+        # This may lead to unexpected results, but said unexpectedness
+        # should be easy to avoid. An example of the unexpected results
+        # that may occur would be if one prefix was `!` and one was `!a`.
+        for prefix in prefixes:
+            if prefix and content.startswith(prefix) and not require_mention:  # don't cut off commands with mentions.
```

Comment:
I don't understand what the check to `require_mention` is doing here? We process mentions out of the message above if `reuqire_mention` is set, so this is presumably unnecessary?

Reply:
I honestly don't know. Maybe I was thinking of the case where `{@mention} {prefix}` is sent when `{prefix}` is a command, but now that I think of it, this case would only be reached by people who willing configure their bots to have this behaviour (set require mention to true and have a prefix).

tl;dr I'll fix it